### PR TITLE
Emag shuttles

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

This here adds new config file (and renames the old one) in which you specify which shuttles can be bought if the comms console has been emagged.
I've locked a few shuttles behind emag in this PR but this is still subject to change.
closes #2828

## Why It's Good For The Game

Bad shuttles are literal banbait right now. This limits them while not completely removing from the game.
Also this adds actual use to emagging the comms console.

## Changelog
:cl:
add: New option to have a shuttle be only buy-able after comms console is emagged.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
good PR.

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 	var/credit_cost = INFINITY
 	var/can_be_bought = TRUE
+	var/syndicate_exclusive = FALSE	//makes you able to buy the shuttle in emaged comms console even if can_be_bought is FALSE
```
Comment (BeloneX) [inline code/datums/shuttles.dm:14]: grammar correction, should be.
//makes you able to buy the shuttle at a emagged comms console even if can_be_bought is FALSE

File: config/Sage/shuttles_unbuyable.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
-##Listing maps here will make the shuttle not available to buy in comms console.
+##Listing maps here will make the shuttle be avaialable to be bought if comms console is emagged
```
Comment (ghost) [inline config/Sage/shuttles_unbuyable.txt]:
```suggestion
##Listing maps here will make the shuttle be available to be bought if the comms console is emagged
```

File: code/game/machinery/computer/communications.dm [hunk 3]
```diff
@@ -566,6 +566,15 @@
 					if(S.prerequisites)
 						dat += "Prerequisites: [S.prerequisites]<BR>"
 					dat += "<A href='?src=[REF(src)];operation=buyshuttle;chosen_shuttle=[REF(S)]'>(<font color=red><i>Purchase</i></font>)</A><BR><BR>"
+				else if(obj_flags & EMAGGED)
+					if(S.syndicate_exclusive && S.credit_cost < INFINITY)
+						dat += "<b> Shuttle supplied by Syndicate, order on your own risk: </b>"
```
Comment (ghost) [inline code/game/machinery/computer/communications.dm]:
```suggestion
						dat += "<b> Shuttle supplied by the Syndicate, order at your own risk: </b>"
```

File: code/datums/shuttles.dm (hunk 1 above)
Comment (ghost) [inline code/datums/shuttles.dm]:
```suggestion
	var/syndicate_exclusive = FALSE	//makes you able to buy the shuttle at an emagged comms console even if can_be_bought is FALSE
```

**Review by ghost — CHANGES_REQUESTED** [review]
Correct the grammar with what i provided.

**Review by Archanial — COMMENTED** [review]
how do I close this window

File: code/datums/shuttles.dm (hunk 1 above)
Comment (Archanial) [reply code/datums/shuttles.dm]: actually "at an emagged"

**Comment by ghost** [comment]
proof that oracle was years ahead of us https://github.com/OracleStation/OracleStation/pull/707/

**Comment by utilbird** [comment]
Emagged consoles probably shouldn’t read out the name of who bought the shuttle. Unless bloodbaths are something you’re aiming for

**Comment by ghost** [comment]
That's kinda on the antag, if they don't wanna get lynched for buying a shitty shuttle then they should get an ID with another name or get gud

**Comment by eeSPee** [comment]
> Emagged consoles probably shouldn’t read out the name of who bought the shuttle. Unless bloodbaths are something you’re aiming for

Agent ID, pal.

**Comment by utilbird** [comment]
@Gregolry @eeSPee lol, the purchaser’s name uses usr.real_name not the ID card

**Comment by PowerfulBacon** [comment]
I think it should be unlocked through screwdrivering the board, so antagonists without emag can buy them, self antaggers who are going to get banned can buy them and you can't accidently buy them thinking the 100k shuttle named hyperfractal gigashuttle is actually really deadly

**Comment by PowerfulBacon** [comment]
Closed in favour of https://github.com/BeeStation/BeeStation-Hornet/pull/2920/files

**Pull request closed without merging**